packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -289,10 +318,13 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
     },
     loadMore() {
       // Ignore if already loading more or if performing server side filtering.
-      if (data.state === 'loadingMore' || data.state === 'filtering' || data.cursor == null) {
+      // `data.state` won't be up to date if multiple loadMore's are called at the same time so we track loadingMore state
+      // via ref. This allows us to properly early return
+      if (loadingMore.current || data.state === 'filtering' || data.cursor == null) {
         return;
       }
 
+      loadingMore.current = true;
```

Comment:
Not very happy about having to track `loadingMore` in a ref. Problem is that `data.state` may not be up to date if someone calls `loadMore` multiple times in quick succession before the `dispatch` call has a chance to update `data.state` from `idle` to `loadingMore`.

Trying to figure out a way to handle this case in the `loadingMore` reducer case above, but so far I've ended up with infinite `loadMore` loops in ComboBox occasionally since the data state will change back and forth from `error` and `loadingMore` repeatedly (Virtualizer useLayoutEffect's loadMore call is only blocked if loadingState is equal to loadingMore).

Reply:
Possible solution would be to do the following:
- call `action.abortController.abort();` in a `loadingMore` case handler for `data.state === loadingMore` in reducer
- modify `ComboBox`/`ListBoxBase`/`Virtualizer` so that `loadingState` is passed through. Add conditional that `loadMore` isn't triggered if `loadingState === error` (not sure about this one)
- fix reducer logic so that the `data.state === idle/error` case with `action.type === error/success` returns an updated `data.state = idle` if action.type is success (aka the original loadMore has finished).

Reply:
new solution in ec9fccf, old solution in ad9cb18. Turns out I didn't have to do most of the above, just needed to make sure loadMore errors maintained the loadMore state rather than changing to an error state